shell/platform/darwin/ios/framework/Source/FlutterEngine.mm

```diff
@@ -493,22 +506,43 @@ - (BOOL)createShell:(NSString*)entrypoint libraryURI:(NSString*)libraryURI {
     FML_LOG(ERROR) << "Could not start a shell FlutterEngine with entrypoint: "
                    << entrypoint.UTF8String;
   } else {
-    [self setupChannels];
-    if (!_platformViewsController) {
-      _platformViewsController.reset(new flutter::FlutterPlatformViewsController());
+    if (!oc_async_init_block) {
+      [self setUpWhenInit];
     }
-    _publisher.reset([[FlutterObservatoryPublisher alloc] init]);
-    [self maybeSetupPlatformViewChannels];
-    _shell->GetIsGpuDisabledSyncSwitch()->SetSwitch(_isGpuDisabled ? true : false);
   }
-
   return _shell != nullptr;
 }
 
+- (BOOL)createShell:(NSString*)entrypoint libraryURI:(NSString*)libraryURI {
+  return [self createShell:entrypoint libraryURI:libraryURI asyncInitCallback:nil];
```

Comment:
It seems to be a breaking change. Developers who're using the original api: createShell:libraryURI would have no knowledge about the changes because the api signature remains while the implementation is changed. One who expect the engine is ready after this api is called would be confused about that behavior.
In this case, a new api would be clear to understand and avoid the confusion.

Reply:
not break change.  
- origin api still is `createShell: entrypoint libaryUri`,
- new api is `createShell: entrypoint libaryUri asyncInitCallback:nil`

Cause origin api impl has too much logic ,so change origne api signature (line 407), and modify code by judge `callback!=null`
For compatible origin api, add same api signature here, called new api  with  params`callback:nill`

for developer , there is nothing change for them . they still can call origin api `createShell: entrypoint libaryUri`,and behavior is same as before